syntax = "proto3";

message SongMessage {
  string title = 2;
  float duration = 3;
}

message CreateSongRequest {
  bytes playlist_id = 1;
  SongMessage song = 2;
}

message CreateSongResponse { bytes song_id = 1; }

message DeleteSongRequest {
  bytes playlist_id = 1;
  bytes song_id = 2;
}

message DeleteSongResponse {}

message GetSongRequest {
  bytes playlist_id = 1;
  bytes song_id = 2;
}

message GetSongResponse {
  bytes song_id = 1;
  SongMessage song = 2;
}

message UpdateSongRequest {
  bytes playlist_id = 1;
  bytes song_id = 2;
  SongMessage song = 3;
}

message UpdateSongResponse {}

service Song {
  rpc CreateSong(CreateSongRequest) returns (CreateSongResponse) {}
  rpc DeleteSong(DeleteSongRequest) returns (DeleteSongResponse) {}
  rpc GetSong(GetSongRequest) returns (GetSongResponse) {}
  rpc UpdateSong(UpdateSongRequest) returns (UpdateSongResponse) {}
}